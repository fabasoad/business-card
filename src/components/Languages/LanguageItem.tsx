import * as React from 'react'
import FlagIconFactory from 'react-flag-icon-css'

export interface LanguageItemProps {
  children?: React.ReactNode
  code: string
}

export default function LanguageItem(props: LanguageItemProps) {
  const FlagIcon = FlagIconFactory(React, { 'useCssModules': false })
  return (
    <div className="mb-4 mx-4">
      <FlagIcon className="mx-auto" code={props.code} size="4x" />
      {props.children}
    </div>
  )
}
