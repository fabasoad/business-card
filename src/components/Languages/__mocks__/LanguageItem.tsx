import * as React from 'react'
import { LanguageItemProps } from '../LanguageItem'

export default function LanguageItem({
  children, code
}: React.PropsWithChildren<LanguageItemProps>) {
  return (
    <div data-testid={code}>{children}</div>
  )
}
