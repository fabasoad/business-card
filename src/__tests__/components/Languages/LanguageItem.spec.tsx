/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import LanguageItem from '../../../components/Languages/LanguageItem'

test('should render LanguageItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <LanguageItem code="gb">test-content</LanguageItem>
  )
  expect(wrapper).toMatchSnapshot()
})
