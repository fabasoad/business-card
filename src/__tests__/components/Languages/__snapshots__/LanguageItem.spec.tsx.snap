// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguageItem correctly 1`] = `
<div
  className="mb-4 mx-4"
>
  <FlagIcon
    className="mx-auto"
    code="gb"
    size="4x"
  />
  test-content
</div>
`;
